include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -175,6 +175,20 @@ class LAPACKFullMatrix : public TransposeTable<number>
   void
   rank1_update(const number a, const Vector<number> &v);
 
+  /**
+   * Apply Givens rotation @p csr (a triplet of cosine, sine and radius)
+   * to this matrix in plane @p i and @p k.
+   * If @p left is <code>true</code>, the rotation is applied from left
+   * and only rows @p i and @p k are affected.
+   * Otherwise, transpose of the rotation matrix is applied from right
+   * and only columns @p i and @p k are affected.
+   */
+  void
+  apply_givens(const std::array<number, 3> &csr,
```

Comment:
> Any particular reason to use a tuple here, rather than three separate arguments?

yes https://www.dealii.org/developer/doxygen/deal.II/namespaceUtilities_1_1LinearAlgebra.html#a22a928b284f7a00ad77fc52bca120835

> In any case, can you document what these three numbers are?

will do and link to this function.